### DISCUSSION TITLE
cfw建议增加多个worker

### DISCUSSION THREAD

**Discussion by DreamUFO** [SeaHOH/GotoX#163] [点子]
如题
cloudflare 的worker 有访问次数限制 100000 次每天
有时候一个worker不够用 建议可以增加多个worker 用逗号隔开，轮询使用worker地址
例如如下配置 
[cfw]
#请填写 worker 的完整域名，只能填写一个
#如 cloudflare 提供的免费域名， worker-name.your-subdomain.workers.dev
#或者用你自己绑定的域名
worker = worker-name.your-subdomain1.workers.dev,worker-name.your-subdomain2.workers.dev

**Comment by SeaHOH** [comment]
这个十万是帐号限制，不是单个 Worker。而且即使你能注册多个帐号，我也不会提供这样的功能，故意的。

**Comment by DreamUFO** [comment]
如果 作者真的是想提供像GAE那样方便的 代理的话 还是有必要提供的
毕竟GAE也是支持添加多个地址的
如果是个人任性，那我也只能支持你了。

**Comment by SeaHOH** [comment]
这可不是任性，而是为大多数用户着想。别人提供服务是有条款的，想使用比免费计划更多的服务，简单：掏钱升级，不要老想着钻空子薅羊毛（实际上已经在轻轻地薅了，但这是官方特许的”达则兼济天下“）。人家毕竟是商业公司，还要赚钱，前阵子还把 reCaptcha 换成更便宜的 hCaptcha。虽然 CloudFlare 因为理念缘故容忍度要高些，但要真被薅破功了，再找下一头羊也麻烦，搞不好都不让你薅呢。

当然，GAE 不同，狗大户，不薅白不薅，但是现在不敢啊，隔壁勇士还在薅，赌谷歌的良心尚未尽墨。

还有啊，一秒一个请求，一天也才八万七不到。如果你有超出常规的需求，最好不要指望免费的服务，你会发现使用过程非常不稳定，还可能特别折腾。

**Comment by DreamUFO** [comment]
感谢你的项目 略作调整 实现了多个worker工作 
不是因为要用来滥用 而是作为一个上网重度用户 10w 只能说是基本满足 20W足矣(我个人的情况，不排除还有正常需求还有更多请求的大兄弟)
作者可以试试 打开一个淘宝或者京东或者类似的外面大点网站首页 有多少次请求
有些站点后台还在不停的请求数据 一分钟请求了很多次 这样的网站比比皆是
还有像 google地图这种网站 地图随意浏览几次 就成千上万次请求
别拿百毒地图来搪塞 有些地方地图很多年都不更新一次还模糊
本不想再回复 只想提醒作者 不清楚真正需求之前 先把别人往好的方面想

**Comment by SeaHOH** [comment]
可能我还是没有讲清楚，这里再说明一下，并非针对特定个人，请你不要误会。

部分用户的正常代理需求确实会比较大，但这并不意味着就是常规普遍的。上面也有讲，这种规模的需求使用免费服务肯定会感受到其不稳定性，这是免费服务定位和用户强大需求的矛盾所决定，这种情况一般是建议使用付费服务，稳定更有保障。

另外，如果发布这个功能，之后用户将怎么使用它，作者是没有办法控制的，一旦有人发现其便利性，肯定会被滥用。这是我多年阅历，绝非凭空臆测，所以我的选择是不提供这种功能，也恳请你不要发布你修改的版本。

**Comment by SeaHOH** [comment]
其实这个工具不光是提供一个 Workers 代理，也有其它功能可帮助突破封锁，使用**自动代理端口**并合理配置，可减少 Workers 代理请求。

1. 自定义跳转链接，可实现强制 https；
1. DoH、自定义 IP 列表 (类似于 Hosts)，可避免 DNS 污染；
1. 伪造 SNI，可直连封锁 SNI 但未封 IP 的站点；
1. 可调用其它代理服务 (HTTP、SOCKS4/5)；
1. 还可以灵活调配自动规则实现一些非常规的特别需求。

**Comment by DreamUFO** [comment]
尊重原创 我不会发布修改版本
我也痛恨少数滥用 給多数人带来了不便。
就像 当年国外大多数PT站都封锁CN IP 一样

**Comment by SeaHOH** [comment]
理解万岁 :handshake:

**Comment by lon91ong** [comment]
> 其实这个工具不光是提供一个 Workers 代理，也有其它功能可帮助突破封锁，使用**自动代理端口**并合理配置，可减少 Workers 代理请求。
> 
> 1. 自定义跳转链接，可实现强制 https；
> 2. DoH、自定义 IP 列表 (类似于 Hosts)，可避免 DNS 污染；
> 3. 伪造 SNI，可直连封锁 SNI 但未封 IP 的站点；
> 4. 可调用其它代理服务 (HTTP、SOCKS4/5)；
> 5. 还可以灵活调配自动规则实现一些非常规的特别需求。

这几种方式如果能配合实例做个说明会方便我这样的小白很多，也能减少重复issue提问的麻烦

**Comment by SeaHOH** [comment]
Readme 大致介绍了主要功能和使用方法，“详细文档”大多都已写在配置文件的注释中，默认配置和规则就是示例。我在配置中添加注释的目的，就是方便用户根据自身需求来修改。如确实有未表述清楚的地方，请随意提问不要怕麻烦，我会根据反馈修改，这样才会越来越完善。

SNI 伪造功能在 wiki 页面有比注释更详细的说明。而非常规的特别需求，只能是熟悉理解掌握后才能做到相应配置。如果当前确有特别需求，但还未熟悉应用无法自行配置，还是请你随意提问，不麻烦。

如还有需要特别说明的功能、概念或使用方法等，我会根据反馈酌情添加到合适的地方。

**Comment by SeaHOH** [comment]
对，这和 GAE 不同，计量机制是请求数，使用它会快速消耗请求数，所以没有使用。

**Comment by SeaHOH** [comment]
难道不该使用低时延的 IP 吗？默认解析结果就能满足一般使用需求。
这个功能是提供给某些有特殊需求的用户，可能会触发导致 CloudFlare 临时 ban 掉你的 IP，一般不使用它。

**Comment by SeaHOH** [comment]
> 这个功能是提供给某些有特殊需求的用户，可能会触发导致 CloudFlare 临时 ban 掉你的 IP，一般不使用它。

站长使用的测试工具就可以用，也有专门的测试网站，但是由客户端强行自定连接的结果就是这样。
